### PR TITLE
It is a extension of var and list

### PR DESCRIPTION
**Pull request by qxsck**
It is a extension of var and list

### CODE REVIEW

**Comment by GarboMuffin** [comment]
I rewrote many of the blocks to avoid issues like:

 - local variabes and lists not working at all
 - getting an item from a list being O(n) instead of O(1)
 - force coercing all variable and list values to strings
 - not always returning a value

etc.

I have not touched getIndexOfList, getIndexsOfList, length, listHaveValue, or copyList yet, not because they don't have issues, but because I don't have time right now. Consider seeing the changes made to the first few functions and make similar changes to the rest.

**Comment by qxsck** [comment]
> 我重写了许多块以避免以下问题：
> 
> * 局部变量和列表根本不起作用
> * 从列表中获取 O（n） 而不是 O（1） 的项目
> * 强制将所有变量和列表值强制转换为字符串
> * 并不总是返回值
> 
> 等。
> 
> 我还没有接触getIndexOfList，getIndexsOfList，length，listHaveValue或copyList，不是因为他们没有问题，而是因为我现在没有时间。请考虑查看对前几个函数所做的更改，并对其余函数进行类似的更改。

**Comment by qxsck** [comment]
> 我重写了许多块以避免以下问题：
> 
> * 局部变量和列表根本不起作用
> * 从列表中获取 O（n） 而不是 O（1） 的项目
> * 强制将所有变量和列表值强制转换为字符串
> * 并不总是返回值
> 
> 等。
> 
> 我还没有接触getIndexOfList，getIndexsOfList，length，listHaveValue或copyList，不是因为他们没有问题，而是因为我现在没有时间。请考虑查看对前几个函数所做的更改，并对其余函数进行类似的更改。

**Comment by qxsck** [comment]
> I rewrote many of the blocks to avoid issues like:
> 
> * local variabes and lists not working at all
> * getting an item from a list being O(n) instead of O(1)
> * force coercing all variable and list values to strings
> * not always returning a value
> 
> etc.
> 
> I have not touched getIndexOfList, getIndexsOfList, length, listHaveValue, or copyList yet, not because they don't have issues, but because I don't have time right now. Consider seeing the changes made to the first few functions and make similar changes to the rest.

I will try my best to modify them. I'm sorry that I didn't understand these methods before

File: extensions/qxsck/var-and-list.js [hunk 1]
```diff
@@ -1,15 +1,53 @@
...
+      listHaveValue: '列表 [LIST] 包括 [VALUE] 吗？',
+      copyList: '将列表 [LIST1] 复制到列表 [LIST2]',
+    }
+});
   class VarAndList {
     getInfo() {
       return {
-        id: 'qxsckvarandlist',
-        name: 'Var and List',
+        id: 'varandlist',
```
Comment (GarboMuffin) [inline extensions/qxsck/var-and-list.js:44]: you still need your username in the id

Comment (qxsck) [reply extensions/qxsck/var-and-list.js:44]: Oh,sorry

**Pull request merged**